specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json

```diff
@@ -3376,15 +3269,36 @@
           "items": {
             "type": "string"
           },
-          "description": "Authorized IP Ranges to kubernetes API server."
+          "title": "The IP ranges authorized to access the Kubernetes API server.",
+          "description": "IP ranges are specified in CIDR format, e.g. 137.117.106.88/29. This feature is not compatible with clusters that use Public IP Per Node, or clusters that are using a Basic Load Balancer. For more information see [API server authorized IP ranges](https://docs.microsoft.com/azure/aks/api-server-authorized-ip-ranges)."
         },
         "enablePrivateCluster": {
           "type": "boolean",
-          "description": "Whether to create the cluster as a private cluster or not."
+          "title": "Whether to create the cluster as a private cluster or not.",
+          "description": "For more details, see [Creating a private AKS cluster](https://docs.microsoft.com/azure/aks/private-clusters)."
         },
         "privateDNSZone": {
           "type": "string",
-          "description": "Private dns zone mode for private cluster. "
+          "title": "The private DNS zone mode for the cluster.",
+          "description": "The default is System. For more details see [configure private DNS zone](https://docs.microsoft.com/azure/aks/private-clusters#configure-private-dns-zone).",
+          "enum": [
+            "system",
+            "none"
+          ],
+          "x-ms-enum": {
```

Comment:
Provided that doing this wasn't breaking, we may want to do it for a few other fields too that are really enums but aren't labelled as such

Reply:
We should also think about using this in `azure-service-operator`?

Reply:
@Porges using what? The `x-ms-enum` field and it's values? Are we not reading it now? Yeah we maybe should.
Although if we want to investigate becoming an autorest backend at some point it'd happen automatically then

Reply:
Resolving this as it wasn't classified as breaking as far as I can tell, and the discussion about ASO isn't relevant to this PR.